boards: beagle: beagleconnect_freedom: Update capabilities

Ayush1325 opened the pull request:
- Add PWM, ADC, FLASH and Radio to docs
- Add PWM, ADC to beagleconnect_freedom.yaml for testing
- Do not enable IEEE802154 2.4g by default.
- Mostly just saner defaults.

Ayush1325 committed:
boards: beagle: beagleconnect_freedom: Add PWM

- Add PWM support to base devicetree

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 committed:
samples: blinky_pwm: Remove enabling pwm in overlay

- Since PWM config has been moved to base devicetree, it can be safely
  removed from the sample.

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 committed:
samples: drivers: adc: Remove ADC node from overlay

- Since ADC node is now enabled in the base dt, this is no longer
  required

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 commented:
cc @vaishnavachath

Ayush1325 commented on boards/beagle/beagleconnect_freedom/beagleconnect_freedom.dts:
```diff
@@ -34,6 +34,11 @@
 		zephyr,code-partition = &slot0_partition;
 	};
 
+	zephyr,user {
```
I see, I added them here because tests fail if `io-channels` is missing. I will move it to test overlays.

Ayush1325 commented on boards/beagle/beagleconnect_freedom/beagleconnect_freedom.dts:
```diff
@@ -228,3 +233,64 @@
+	#size-cells = <0>;
+
+	/* MB1 AN */
+	channel@9 {
```
Yes, they are available as pins to be used  by user. 2 other channels are also technically present, but they are being used internally for antenna and thus not available for the user.

Ayush1325 commented on boards/beagle/beagleconnect_freedom/beagleconnect_freedom.dts:
```diff
@@ -206,10 +206,6 @@
-&ieee802154 {
-	status = "okay";
-};
-
```
A lot of zephyr networking samples (zperf, http_server, etc) will complain if both 2.4g and subg ieee802154 are enabled simultaneously.
This leads to a lot of new people getting confused with the network code not working without any clear errors.
As you can see, we already set the alias `zephyr,ieee802154 = &ieee802154g;`, so I think it would be best to enable only subg by default.

henrikbrixandersen replied:
Okay. Perhaps add this as a separate commit with the above explanation?

Ayush1325 committed:
boards: beagle: beagleconnect_freedom: Add ADC

- Enable ADC node to the base board.

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 committed:
tests: drivers: adc: adc_api: Add beagleconnect_freedom overlay

- Enabling adc testing for beagleconnect_freedom

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 committed:
boards: beagle: beagleconnect_freedom: Disable 2.4g ieee802154

- A lot of zephyr networking samples (zperf, http_server, etc) will
  complain if both 2.4g and subg ieee802154 are enabled simultaneously.
- This leads to a lot of new people getting confused with the network code
  not working without any clear errors.
- As we already set the alias `zephyr,ieee802154 = &ieee802154g`, I think
  it would be best to enable only subg by default.

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 committed:
boards: beagle: beagleconnect_freedom: doc: Update capabilities

- Add PWM, ADC, FLASH and Radio to docs

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 committed:
boards: beagle: beagleconnect_freedom: Enable adc, pwm tests

- Enable testing for ADC and PWM.

Signed-off-by: Ayush Singh <ayush@beagleboard.org>
